### PR TITLE
hover support

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
After #1009 

Supports `textDocument/hover`

### CODE REVIEW

File: src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        powerShellContext);
+
+                    if (commandInfo != null)
+                    {
+                        symbolDetails.Documentation =
+                            await CommandHelpers.GetCommandSynopsisAsync(
+                                commandInfo,
+                                powerShellContext);
+
+                        if (commandInfo.CommandType == CommandTypes.Application)
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs]: My preference here would be to `return` early. Also to put the bigger `case` at the bottom

File: src/PowerShellEditorServices.Engine/Services/Symbols/SymbolsService.cs [hunk 2]
```diff
@@ -227,5 +229,38 @@ public List<SymbolReference> FindSymbolsInFile(ScriptFile scriptFile)
...
+        {
+            SymbolReference symbolReference =
+                AstOperations.FindSymbolAtPosition(
+                    scriptFile.ScriptAst,
+                    lineNumber,
+                    columnNumber);
+
+            if (symbolReference == null)
+            {
+                // TODO #21: Return Result<T>
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/Symbols/SymbolsService.cs]: What does this TODO mean? Is it obsolete now?

File: src/PowerShellEditorServices.Engine/Services/TextDocument/Handlers/HoverHandler.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+
+namespace PowerShellEditorServices.Engine.Services.Handlers
+{
+    public class HoverHandler : IHoverHandler
+    {
+        private readonly DocumentSelector _documentSelector = new DocumentSelector(
+            new DocumentFilter()
+            {
+                Pattern = "**/*.ps*1"
```
Comment (rjmholt) [inline src/PowerShellEditorServices.Engine/Services/TextDocument/Handlers/HoverHandler.cs]: We should factor this out into a util class

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/TextDocument/Handlers/HoverHandler.cs]: yep on the list but skipping it for now

File: src/PowerShellEditorServices.Engine/Services/Symbols/SymbolsService.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/Symbols/SymbolsService.cs]: https://github.com/PowerShell/PowerShellEditorServices/issues/21 yep

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/Symbols/SymbolsService.cs]: deleted

File: src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs]: refactored

File: src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                case SymbolType.Parameter:
+                    // TODO: Get parameter help
+                    symbolDetails.DisplayString = "(parameter) " + symbolReference.SymbolName;
+                    return symbolDetails;
+
+                case SymbolType.Variable:
+                    symbolDetails.DisplayString = symbolReference.SymbolName;
+                    return symbolDetails;
+
+                default:
```
Comment (rkeithhill) [inline src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs]: Might be nice to start thinking about support for class and enum?  Maybe it's a simple here as returning a string of `class foo` and `enum bar`?

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices.Engine/Services/Symbols/SymbolDetails.cs]: I like that - lets defer the work to a later PR when everything gets hooked up.

File: src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs
Comment (Neaykarn181063) [inline src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs:1]: @src/PowerShellEditorServices.Engine/LanguageServer/OmnisharpLanguageServer.cs

**Review by Neaykarn181063 — COMMENTED** [review]
💕

**Pull request merged**